nexus/pack.cc

```diff
@@ -2075,6 +2075,239 @@ struct NexusPacker
+                        {
+                            log_error("Cannot have %s.TSDATA0 and %s.T driven by different nets (%s vs. %s)\n",
+                                ctx->nameOf(iol), ctx->nameOf(iob),
+                                ctx->nameOf(iol_ti), ctx->nameOf(iob_t));
+                        }
+
+                        // Re-connect TSDATA (even if it has already been
+                        // connected to gnd_net, see the condition above).
+                        if (!iol->ports.count(id_TSDATA0)) {
+                            iol->addInput(id_TSDATA0);
+                        }
+                        disconnect_port(ctx, iol, id_TSDATA0);
+                        connect_port(ctx, iob_t, iol, id_TSDATA0);
+
+                        if (ctx->debug) {
+                            log_info("Reconnecting %s.TSDATA0 to %s\n", ctx->nameOf(iol), ctx->nameOf(iob_t));
+                        }
+
+                        // Check if the T input is driven by a flip-flop. Store
+                        // in the map.
+                        CellInfo* ff = get_ff_for_iob(iob, id_T, iol);
+                        if (ff != nullptr) {
+                            tff_map[ff->name].push_back(std::make_pair(
+                                iol->name, iob->name));
+                        }
+                    }
+                }
+            }
+        }
+
+        // Integrate flip-flops that drive T with IOLOGIC
+        for (auto& it : tff_map) {
+            CellInfo* ff = ctx->cells.at(it.first).get();
+
+            NetInfo* ff_d = get_net_or_empty(ff, id_M); // FIXME: id_D or id_M ?!
+            NPNR_ASSERT(ff_d != nullptr);
+
+            NetInfo* ff_q = get_net_or_empty(ff, id_Q);
+            NPNR_ASSERT(ff_q != nullptr);
+
+            log_info("FF '%s'\n", ctx->nameOf(ff));
+            for (auto& it : ff->params) {
+                log_info(" '%s'='%s'\n", it.first.c_str(ctx), it.second.as_string().c_str());
+            }
+
+            for (auto& ios : it.second) {
+                CellInfo* iol = ctx->cells.at(ios.first).get();
+                CellInfo* iob = ctx->cells.at(ios.second).get();
+
+                log_info("Integrating %s into %s\n", ctx->nameOf(ff), ctx->nameOf(iol));
+
+                // Disconnect "old" T net
+                disconnect_port(ctx, iol, id_TSDATA0);
+                disconnect_port(ctx, iob, id_T);
+
+                // Connect the "new" one
+                connect_port(ctx, ff_d, iol, id_TSDATA0);
+                connect_port(ctx, ff_d, iob, id_T);
+
+                // Propagate parameters
+                iol->params[id_SRMODE] = ff->params.at(id_SRMODE);
+                iol->params[id_REGSET] = ff->params.at(id_REGSET);
+
+                iol->params[ctx->id("TSREG.REGSET")] = std::string("SET");
+
+                // CEOUTMUX.1
+                iol->params[ctx->id("CEOUTMUX")] = std::string("1");
+
+                // FIXME: Workaround for an unknown bit
```

Comment:
I'll try and have a poke at the fuzzer later today.

Reply:
unknown bit should be added in https://github.com/gatecat/prjoxide/commit/7d56d6392d859e18155e43818a62ea3f5789946a

Reply:
Great, thanks. This unknown bits I use seem to control whether T is driven by `IOLOGIC` or externally.

Reply:
I'll update nextpnr to use correct FASM features now.